lib/secure-boot-lib-ami.robot

```diff
@@ -0,0 +1,228 @@
+*** Settings ***
+Documentation       Collection of Dasharo keywords related to UEFI Secure Boot
+
+Resource            ./secure-boot-lib-common.robot
+
+
+*** Variables ***
+${RESET_KEYS_OPTION}=           > Restore Factory Keys
+${INCORRECT_FORMAT_MESSAGE}=    Load Error
+
+
+*** Keywords ***
+Get Secure Boot Menu Construction
+    [Documentation]    Secure Boot menu is very different than all
+    ...    of the others so we need a special keyword for parsing it.
+    ...    Return only selectable entries. If some menu option is not
+    ...    selectable (grayed out) it will not be in the menu construction
+    ...    list.
+    [Arguments]    ${checkpoint}=ESC: Exit    ${lines_top}=1    ${lines_bot}=1
+    ${out}=    Read From Terminal Until    ${checkpoint}
+    # At first, parse the menu as usual
+    ${menu}=    Parse Menu Snapshot Into Construction    ${out}    ${lines_top}    ${lines_bot}
+    Remove Values From List    ${menu}    Not Active    Active
+    RETURN    ${menu}
+
+Enter Secure Boot Menu
+    [Documentation]    This keyword enters Secure Boot menu after the platform was powered on.
+    Enter Setup Menu
+    Read From Terminal Until    ---/
+    Press Key N Times    1    ${ARROW_RIGHT}
+    Press Key N Times    1    ${ARROW_RIGHT}
+    Press Key N Times    1    ${ARROW_RIGHT}
+    Press Key N Times    1    ${PAGEDOWN}
+    Read From Terminal
+    Press Enter
+
+Enter Secure Boot Menu And Return Construction
+    [Documentation]    This keyword enters Secure Boot menu after the platform was powered on. Returns Secure Boot menu construction.
+    Enter Secure Boot Menu
+    ${sb_menu}=    Get Secure Boot Menu Construction
+    ${sb_menu}=    Get Slice From List    ${sb_menu}    1
+    RETURN    ${sb_menu}
+
+Enter Key Management And Return Construction
+    [Documentation]    Enters (Advanced) Key Management menu and returns constructions.
+    ...    Should be called from secure boot menu
+    # robocop: disable=unused-argument
+    [Arguments]    ${sb_menu}=${EMPTY}
```

Comment:
@m-iwanicki please correctly me if I am wrong, but I think that we wanted to use the same name of this KWD for Dasharo (https://github.com/Dasharo/open-source-firmware-validation/blob/uefi-sb-release/lib/secure-boot-lib-dasharo.robot#L68) and AMI (https://github.com/Dasharo/open-source-firmware-validation/blob/uefi-sb-release/lib/secure-boot-lib-ami.robot#L46) while they work differently. Plus sometimes this keyword is triggered with argument (https://github.com/Dasharo/open-source-firmware-validation/blob/uefi-sb-release/dasharo-security/secure-boot.robot#L490) and other times is not (https://github.com/Dasharo/open-source-firmware-validation/blob/uefi-sb-release/dasharo-security/secure-boot.robot#L326). Maybe it could be possible to get rid off this argument at all. @m-iwanicki WDYT?

Reply:
@TomaszAIR Yes in some tests this keyword is used with arguments (on Dasharo) but it's not needed on AMI BIOS.
When this keyword is used with argument then that means before using it there was used keyword that returned menu construction. Either we would need to change that or I guess we could always `Reenter Menu` to make sure we can read it from Terminal.